libs/geodata/riversurvey/geodatariversurveycrosssectionwindowgraphicsview.cpp

```diff
@@ -1159,15 +1159,15 @@ void GeoDataRiverSurveyCrosssectionWindowGraphicsView::inspectLimits(bool* minli
 		left = 0;
 		*minlimit = false;
 	}
-	if (right == alist.count()) {
-		right = alist.count() - 1;
+	if (right == alist.size()) {
+		right = alist.size() - 1;
 		*maxlimit = false;
 	}
 	double min1, min2, max1, max2;
 	min1 = alist[left].position() - alist[selectedPoints.first().row()].position();
 	max1 = alist[right].position() - alist[selectedPoints.last().row()].position();
-	min2 = 0 - alist.last().position();
-	max2 = 0 - alist.first().position();
+	min2 = 0 - (*(alist.begin() + alist.size() - 1)).position();
+	max2 = 0 - (*(alist.begin())).position();

```

Comment:
I think this is easier to read (for lines 1169 and 1170):
	min2 = 0 - alist.front().position();
	max2 = 0 - alist.back().position();